### PR TITLE
Improvement to strings using label

### PR DESCRIPTION
**Pull request by ehom**
Hi @jbondpdx &mdash; please review the propose changes in issues.rb only.

Thanks!

### CODE REVIEW

File: lib/puppet/pops/issues.rb [hunk 1]
```diff
@@ -143,7 +143,7 @@ def self.hard_issue(issue_code, *args, &block)
   # @todo configuration
   #
   NAME_WITH_HYPHEN = issue :NAME_WITH_HYPHEN, :name do
-    _("%{issue} may not have a name containing a hyphen. The name '%{name}' is not legal") % { issue: label.a_an_uc(semantic), name: name }
+    _("The name '%{name}' contains a hyphen which is illegal in an expression of type %{type}.") % { type: label(semantic), name: name }
```
Comment (jbondpdx) [inline lib/puppet/pops/issues.rb]: should have a comma after "hyphen"

File: lib/puppet/pops/issues.rb [hunk 2]
```diff
@@ -153,28 +153,28 @@ def self.hard_issue(issue_code, *args, &block)
...
+    _("A variable name may not contain a hyphen. The name '%{name}' is not legal.") % { name: name }
   end
 
   # A class, definition, or node may only appear at top level or inside other classes
   # @todo Is this really true for nodes? Can they be inside classes? Isn't that too late?
   # @api public
   #
   NOT_TOP_LEVEL = hard_issue :NOT_TOP_LEVEL do
-    _("Classes, definitions, and nodes may only appear at toplevel or inside other classes")
+    _("Classes, definitions, and nodes may only appear at the top level or inside other classes.")
```
Comment (jbondpdx) [inline lib/puppet/pops/issues.rb]: move "only" after "appear" (...nodes may appear only at the top level or...)

File: lib/puppet/pops/issues.rb [hunk 3]
```diff
@@ -153,28 +153,28 @@ def self.hard_issue(issue_code, *args, &block)
...
   end
 
   NOT_ABSOLUTE_TOP_LEVEL = hard_issue :NOT_ABSOLUTE_TOP_LEVEL do
-    _("%{value} may only appear at toplevel") % { value: label.a_an_uc(semantic) }
+    _("An expression of type %{expression_type} may only appear at the top level.") % { expression_type: label(semantic) }
```
Comment (jbondpdx) [inline lib/puppet/pops/issues.rb]: move "only" to after "appear" (same construction as above)

File: lib/puppet/pops/issues.rb [hunk 4]
```diff
@@ -347,20 +347,20 @@ def self.hard_issue(issue_code, *args, &block)
   # E.g. an arithmetic expression where a hostname is expected.
   #
   ILLEGAL_EXPRESSION = hard_issue :ILLEGAL_EXPRESSION, :feature, :container do
-    _("Illegal expression. %{expression} is unacceptable as %{feature} in %{container}") % { expression: label.a_an_uc(semantic), feature: feature, container: label.a_an(container) }
+    _("Illegal expression: %{expression}. It is unacceptable as %{feature} in an expression of type %{container}.") % { expression: label(semantic), feature: feature, container: label(container) }
```
Comment (jbondpdx) [inline lib/puppet/pops/issues.rb]: I think it's clearer without the sentence break. Suggest removing ". It"

File: lib/puppet/pops/issues.rb [hunk 5]
```diff
@@ -502,7 +502,7 @@ def self.hard_issue(issue_code, *args, &block)
   end
 
   CAPTURES_REST_NOT_SUPPORTED = hard_issue :CAPTURES_REST_NOT_SUPPORTED, :container, :param_name do
-    _("Parameter $%{param} has 'captures rest' - not supported in %{container}") % { param: param_name, container: label.a_an(container) }
+    _("Parameter $%{param} has 'captures rest' which is not supported in an expression of type %{container}.") % { param: param_name, container: label(container) }
```
Comment (jbondpdx) [inline lib/puppet/pops/issues.rb]: should have a comma after "rest'": (...'captures rest', which...)

File: lib/puppet/pops/issues.rb [hunk 6]
```diff
@@ -894,15 +901,15 @@ def self.hard_issue(issue_code, *args, &block)
   end
 
   EXPRESSION_NOT_SUPPORTED_WHEN_SCRIPTING = issue :EXPRESSION_NOT_SUPPORTED_WHEN_SCRIPTING, :klass do
-    _("%{expr} is only available when compiling a catalog") % { expr: label.a_an_uc(klass) }
+    _("An expression of type %{expression_type} is only available when compiling a catalog.") % { expression_type: label(klass) }
```
Comment (jbondpdx) [inline lib/puppet/pops/issues.rb]: move "only" to after "available": ...is available only when...

File: lib/puppet/pops/issues.rb (hunk 5 above)
Comment (ehom) [reply lib/puppet/pops/issues.rb]: recommendation accepted.

File: lib/puppet/pops/issues.rb (hunk 6 above)
Comment (ehom) [reply lib/puppet/pops/issues.rb]: updated in latest commit.

File: lib/puppet/pops/issues.rb (hunk 1 above)
Comment (ehom) [reply lib/puppet/pops/issues.rb]: updated in the latest commit.

File: lib/puppet/pops/issues.rb (hunk 4 above)
Comment (ehom) [reply lib/puppet/pops/issues.rb]: updated in latest commit.

File: lib/puppet/pops/issues.rb (hunk 2 above)
Comment (ehom) [reply lib/puppet/pops/issues.rb]: updated in latest commit.

**Comment by ehom** [comment]
superceded by this pull request: https://github.com/puppetlabs/puppet/pull/7390

**Pull request closed without merging**